_includes/seo.html

```diff
@@ -4,14 +4,14 @@
 {%- endif -%}
 {%- assign seo_url = seo_url | default: site.github.url -%}
 
-{% assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' %}
+{% assign title_separator = site.title_separator | default: '-' %}
 
 {%- if page.title -%}
   {%- assign seo_title = page.title | append: " " | append: title_separator | append: " " | append: site.title -%}
 {%- endif -%}
 
 {%- if seo_title -%}
-  {%- assign seo_title = seo_title | markdownify | strip_html | strip_newlines | escape_once -%}
```

Comment:
Try applying these filters on `page.title` a few lines above before appending the spaces, `title_separator` and `site.title`.